composed-true-concidered-harmful.md

```diff
@@ -1,35 +1,583 @@
 
-So, for example, a [click event](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event) in a simple HTML document will originate at the document boundary (`document`), capture down the DOM tree before arriving at the dispatching element (often an `a` or `button`), and then subsequently bubbling back up the DOM tree until returning to the `document`. This is due to their being no `document-fragments` to worry about and the native `click` event having `bubbles` set to `true` by default. Bubbling events acting in this way is something that developers have come to expect from the web in that our pages have been single document realities since the beginning of the DOM. However, with the advent of [Shadow DOM](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM), and our ability to add it to both [Custom Elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements), as well as many of the DOM elements we've worked with since well before the [Web Component](https://developer.mozilla.org/en-US/docs/Web/Web_Components) specifications came into reality, we now have the possibility of a `document` that consists of many `document-fragments`.
+### The Composed Path Less Traveled
 
-When a click event is dispatched inside of a `document-fragment` as provided by a `shadowRoot` then the capture phase of that event will start from the `document-fragment` unless `composed` is set to `true`. Equally, the bubble phase (when `bubbles` is set to `true`) will end at that same `document-fragment` while the value of `composed` is not `true`. However, due to the fact that a `shadowRoot` is meant to encapsulate the DOM and scripting that goes on within it, when an event has a `composed` value of true, the capture and bubble phases will list the `target` of the event as the closest of the dispatching element or and element with a `shadowRoot` attached. That is to say, regarding the following HTML, that `click` events on the `<a>` will register a target of `<child-with-shadow-root>` when listened for on the `<parent-with-shadow-root>` element:
+There are times when we need a look into that dirty laundry to get a peek at just where that event came from, be it `<button>`, `<div>`, `<a>`, or something else (it's hopefully a `<button>` or `<a>`...a11y people!), and for those times we've got the `composedPath()` method on our events. At any point in the event's lifecycle, calling `composedPath()` on that event will give you an array of all the DOM elements on which it will be heard. The array is listed in "bubble" order, so the zeroeth item will be the dispatching element and the last item will be the last element through which the event will pass. That means you can always use the following code to ascertain the original dispatching element and outline the path along which the event will trave, assuming the previous example HTML:
```

Comment:
```suggestion
There are times when we need a look into that dirty laundry to get a peek at just where that event came from, be it `<button>`, `<div>`, `<a>`, or something else (it's hopefully a `<button>` or `<a>`...a11y, people!), and for those times we've got the `composedPath()` method on our events. At any point in the event's lifecycle, calling `composedPath()` on that event will give you an array of all the DOM elements on which it will be heard. The array is listed in "bubble" order, so the zeroeth item will be the dispatching element and the last item will be the last element through which the event will pass. That means you can always use the following code to ascertain the original dispatching element and outline the path along which the event will trave, assuming the previous example HTML:
```